service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java

```diff
@@ -75,6 +77,8 @@ public ResponseEntity<ApiCreatedControlledAzureDisk> createAzureDisk(
       UUID workspaceId, ApiCreateControlledAzureDiskRequestBody body) {
     features.azureEnabledCheck();
 
+    ResourceValidationUtils.validateCloningInstructions(
```

Comment:
Since this method should be called every time you create a resource, could this call be in `WsmResource.validate()` instead? We already call `validate()` on subclasses (which always call the base class validation method) as part of every resource type's `build()` method, and this matches other static validation we do (like name/description checks) that live in that method.

Reply:
Capital idea!